charts/hivemq-platform/tests/monitoring/hivemq_grafana_dashboard_test.yaml

```diff
@@ -0,0 +1,9631 @@
+
+  - it: with invalid Dashboard namespace, schema validation fails
+    set:
+      monitoring:
+        enabled: true
+        dashboard:
+          namespace: this-is-an-extremely-very-very-long-hivemq-platform-namespace-name
+    asserts:
+      - failedTemplate: {}
+
+  - it: with default values, default ConfigMap resource created
+    set:
+      monitoring.enabled: true
+    asserts:
+      - containsDocument:
+          apiVersion: v1
+          kind: ConfigMap
+          name: hivemq-platform-dashboard-test-hivemq-platform
+
+  - it: with monitoring enabled and default values, ConfigMap metadata created with defaults values
+    set:
+      monitoring.enabled: true
+    asserts:
+      - isSubset:
+          path: metadata
+          content:
+            labels:
+              helm.sh/chart: hivemq-platform-0.0.1
+              app.kubernetes.io/name: hivemq-platform
+              app.kubernetes.io/instance: test-hivemq-platform
+              app.kubernetes.io/version: 1.0.0
+              app.kubernetes.io/managed-by: Helm
+              grafana_dashboard: "1"
+            name: hivemq-platform-dashboard-test-hivemq-platform
+            namespace: test-hivemq-platform-namespace
+
+  - it: with monitoring enabled and custom metadata values, ConfigMap metadata created with custom values
+    release:
+      name: custom-release-name
+      namespace: custom-release-namespace
+    chart:
+      appVersion: 1.0.0
+      version: 0.1.0
+    set:
+      monitoring:
+        enabled: true
+        dashboard:
+          namespace: my-custom-dashboard-namespace
+    asserts:
+      - isSubset:
+          path: metadata
+          content:
+            labels:
+              helm.sh/chart: hivemq-platform-0.1.0
+              app.kubernetes.io/name: hivemq-platform
+              app.kubernetes.io/instance: custom-release-name
+              app.kubernetes.io/version: 1.0.0
+              app.kubernetes.io/managed-by: Helm
+              grafana_dashboard: "1"
+            name: hivemq-platform-dashboard-custom-release-name
+            namespace: my-custom-dashboard-namespace
+
+  - it: with monitoring enabled and overrideDashboard set, custom ConfigMap spec created with custom values
+    set:
+      monitoring:
+        enabled: true
+        dashboard:
+          overrideDashboard: |-
+            {
+              "my-custom-grafana-title": "foobar"
+            }
+    asserts:
+      - exists:
+          path: data["hivemq-dashboard.json"]
+      - equal:
+          path: data["hivemq-dashboard.json"]
+          value: |
+            {
+              "my-custom-grafana-title": "foobar"
+            }
+
+  - it: with monitoring enabled and default values, ConfigMap spec created with defaults values
+    set:
+      monitoring.enabled: true
+    asserts:
+      - exists:
+          path: data["hivemq-dashboard.json"]
+      - equal:
+          path: data["hivemq-dashboard.json"]
+          value: |
+            {
```

Comment:
Uff :sweat_smile: Maybe we can persist a snapshot for this and just check that the value hasn't changed? https://github.com/helm-unittest/helm-unittest/blob/main/README.md#snapshot-testing

Then we wouldn't need to inline the whole JSON again and would only need to update the snapshow when the dashboard JSON is updated.

Reply:
We could also file a feature request to verify against a file, e.g.:
```
valueFrom:
  file: "files/grafana-dashboard.json"
```

Reply:
That's pretty awesome! Thanks for the suggestion, didn't know that one.
Will file a feature request as well.

Reply:
I think we need to check in these files as well, otherwise the assertion will do nothing when executed for the first time (e.g. on each GitHub test execution):
```
        charts/hivemq-platform-operator/tests/__snapshot__/
        charts/hivemq-platform/tests/monitoring/__snapshot__/
```

Reply:
And I think we have to be careful to not change the test name/description, since that seems to be the key in the snapshot files.

Reply:
It kind of makes sense. The assertion is "the value of `path: data["hivemq-dashboard.json"]` must match the snapshot of `it: with monitoring enabled and default values, ConfigMap spec created with defaults values`".

And the snapshot file matches the testsuite file name and contains snapshots for each test like:
```yaml
with monitoring enabled and default values, ConfigMap spec created with defaults values:
  1: |
    ...
```

Reply:
I've added the two files and also fixed the branch protection rules, so the PR can become green.

Reply:
https://github.com/helm-unittest/helm-unittest/issues/500

Reply:
Thanks! I forgot to include the `.snap` files